src/Extensions/Abstractions/Activities/TimedScopesExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.Abstractions.Activities
+{
+	/// <summary>
+	/// Extnesions to TimedScopes class to set additional information
+	/// </summary>
+	public static class TimedScopesExtensions
+	{
+		/// <summary>
+		/// Set result
+		/// </summary>
+		/// <remarks>This property won't be transfered to child activity or via web requests</remarks>
+		public static TimedScope SetResult(this TimedScope timedScope, TimedScopeResult result)
+		{
+			timedScope.Activity.AddTag(ActivityTagKeys.Result, ActivityResultStrings.ResultToString(result));
+			return timedScope;
```

Comment:
What is the reason for also returning the timedscope here, as long as it is not immutable because the change is done on the existing instance anyway?

Reply:
It's done to follow the same fluent API style that `Activity.AddTag` or `Activity.AddBaggage` is using. It's just a convenience that allows changing configuration like:
```csharp
TimedScope = provider.Create(definition)
    .SetResult(Success)
    .SetSubType("ImportantSubType")
    .SetMethadata("SuperUsefulInfo");
```

Reply:
Ah, right.